### PR TITLE
Add map api route

### PR DESCRIPTION
**Pull request by UncleBu1728**
Added `map.ts` under `map/route` containing the skeleton of the routers. This initial implementation includes the router structure without any functions implemented.

### CODE REVIEW

File: src/api/map/routes/map.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { procedure, router } from "../../trpc";
+import { authUser } from "../../auth";
+
+const get = procedure
+  .use(authUser("Mentee"||"Mentor"))
```
Comment (weihanwang) [inline src/api/map/routes/map.ts:5]: no need for "Mentee"||"Mentor". every logged-in user should be able to use the map.

File: src/api/map/routes/map.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  .query(async () =>
+{
+});
+
+
+const list = procedure
```
Comment (weihanwang) [inline src/api/map/routes/map.ts:11]: let's start with this single API function and remove all others. let's implement map reading end-to-end first before moving on to updating etc.

File: src/api/map/routes/map.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  .use(authUser("Mentee"||"Mentor"))
+  .query(async () =>
```
Comment (weihanwang) [inline src/api/map/routes/map.ts:13]: can you write out input parameters and return type?

**Comment by UncleBu1728** [comment]
Vercel is showing errors due to the Eslint rules. Do we need another PR to disable these rules for now? Or work on more API implementation?
image

File: src/api/map/routes/map.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { procedure, router } from "../../trpc";
+import { authUser } from "../../auth";
+import { z } from "zod";
+
+const list = procedure
+  .use(authUser())
+  .input(z.string())
```
Comment (weihanwang) [inline src/api/map/routes/map.ts:7]: an enum type should be explicitly defined. refer to the Role type.

File: src/api/map/routes/map.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  .output(z.array(z.object({})))
```
Comment (weihanwang) [inline src/api/map/routes/map.ts:8]: a struct type should be explicitly defined, with fields like ”定义“、“进阶” etc.

**Review by weihanwang — CHANGES_REQUESTED** [review]

File: src/api/map/routes/map.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable require-await */
+import { procedure, router } from "../../trpc";
+import { authUser } from "../../auth";
+import { z } from "zod";
+import { zRoles } from "shared/Role";
+
+const zLandmark = z.object({
```
Comment (weihanwang) [inline src/api/map/routes/map.ts:8]: this needs to be in a folder referencable by the front end.

File: src/api/map/routes/map.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  层级: z.array(z.string()),
+  相关地标: z.array(z.string()),
+  工具箱: z.string(),
+});
+
+//type Landmark = z.TypeOf<typeof zLandmark>;
+
+const list = procedure
+  .use(authUser())
+  .input(z.object({
```
Comment (weihanwang) [inline src/api/map/routes/map.ts:21]: the input doesn't need any of these fields. it instead needs lattitude which should be an enum.

File: src/api/map/routes/map.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+/* eslint-disable require-await */
+import { procedure, router } from "../../trpc";
+import { authUser } from "../../auth";
+import { z } from "zod";
+import { zRoles } from "shared/Role";
+
+const zLandmark = z.object({
+  定义: z.string(),
+  经度: z.string(),
+  纬度: z.string(),
```
Comment (weihanwang) [inline src/api/map/routes/map.ts:11]: These should be enums

File: src/api/map/routes/map.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  层级: z.array(z.string()),
+  相关地标: z.array(z.string()),
+  工具箱: z.string(),
```
Comment (weihanwang) [inline src/api/map/routes/map.ts:14]: this should be optional().

File: src/api/map/routes/map.ts [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+import { authUser } from "../../auth";
+import { z } from "zod";
+import { zRoles } from "shared/Role";
+
+const zLandmark = z.object({
+  定义: z.string(),
+  经度: z.string(),
+  纬度: z.string(),
+  层级: z.array(z.string()),
+  相关地标: z.array(z.string()),
```
Comment (weihanwang) [inline src/api/map/routes/map.ts:13]: this should be optional

File: src/shared/Map.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
+import { z } from "zod";
+
+export const Longtitude = [
```
Comment (weihanwang) [inline src/shared/Map.ts:3]: `Longtitudes`

File: src/shared/Map.ts [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    "情感",
+    "智慧",
+    "实践"
+] as const;
+
+export const zLongtitude = z.enum(Longtitude);
+
+export const Latitude = [
```
Comment (weihanwang) [inline src/shared/Map.ts:11]: ditto

**Review by weihanwang — CHANGES_REQUESTED** [review]

File: src/shared/Map.ts [hunk 13]
```diff
@@ -1,25 +1,25 @@
...
 
-export const Longtitude = [
+export const Longtitudes = [
     "情感",
     "智慧",
     "实践"
 ] as const;
 
-export const zLongtitude = z.enum(Longtitude);
+export const zLongtitudes = z.enum(Longtitudes);
```
Comment (weihanwang) [inline src/shared/Map.ts:9]: keep it as `zLongtitude`. singular.

File: src/shared/Map.ts [hunk 14]
```diff
@@ -1,25 +1,25 @@
...
 
-export const Latitude = [
+export const Latitudes = [
     "个人成长",
     "事业发展",
     "社会责任"
 ] as const;
 
-export const zLatitude = z.enum(Latitude);
+export const zLatitudes = z.enum(Latitudes);
```
Comment (weihanwang) [inline src/shared/Map.ts:17]: ditto.

File: src/api/map/routes/map.ts [hunk 15]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable require-await */
```
Comment (weihanwang) [inline src/api/map/routes/map.ts:2]: Use per-line statement and not per-file. Move them to immediately above the lines where lint rules are violated.

**Review by weihanwang — APPROVED** [review]

**Pull request merged**